### PR TITLE
Add workspace-scoped events endpoint

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_ee11da30dc16481a9ced3e77` (agent: `codex`).

### Task
Implement a small PRD-backed observability follow-up for AWF: add GET /v1/workspaces/{workspace_id}/events as a workspace-details convenience endpoint for the future web console. Strict TDD: first add failing API tests under tests/unit/api that cover returning newest-first events for one workspace, returning an empty list for a known workspace with no extra events beyond creation when filtered appropriately if needed, 404 for an unknown workspace, and limit bound validation matching /v1/events. Then implement the route using existing WorkspaceEventResponse / WorkspaceEventListResponse shapes, without breaking the existing /v1/events endpoint. Keep the response cursor-ready: { items, next_cursor: null, has_more: false }. Run validation exactly: uv run ruff check src/awf/api src/awf/db tests/unit/api; uv run mypy src/awf/api src/awf/db; uv run pytest tests/unit/api -q. Commit with a conventional commit. Do not push; AWF owns push, PR creation, monitoring, and merge. Do not switch branches; AWF owns branch lifecycle.

---
Validation: 6 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: src/awf/api/routes/workspaces.py [hunk 1]
```diff
@@ -211,6 +212,31 @@ async def list_workspace_overview(
...
 
+@router.get("/{workspace_id}/events", response_model=WorkspaceEventListResponse)
+async def list_workspace_events(
+    workspace_id: str,
+    event_type: Annotated[str | None, Query(min_length=1, max_length=64)] = None,
+    limit: Annotated[int, Query(ge=1, le=500)] = 50,
+    session: AsyncSession = Depends(get_db_session),
+) -> WorkspaceEventListResponse:
+    workspace_repo = WorkspaceRepository(session)
+    workspace = await workspace_repo.get(workspace_id)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/api/routes/workspaces.py]: fixed in commit 42d17ef98c9ce0001f83161871c98554a913a1ae

File: src/awf/api/routes/workspaces.py [hunk 2]
```diff
@@ -211,6 +212,31 @@ async def list_workspace_overview(
...
 
 
+@router.get("/{workspace_id}/events", response_model=WorkspaceEventListResponse)
+async def list_workspace_events(
+    workspace_id: str,
+    event_type: Annotated[str | None, Query(min_length=1, max_length=64)] = None,
+    limit: Annotated[int, Query(ge=1, le=500)] = 50,
+    session: AsyncSession = Depends(get_db_session),
+) -> WorkspaceEventListResponse:
+    workspace_repo = WorkspaceRepository(session)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/api/routes/workspaces.py]: fixed in commit 19dab63cbb3be88cbeaff84d7fcad22446e4c581

File: tests/unit/api/test_events.py [hunk 3]
```diff
@@ -86,3 +111,66 @@ async def test_returns_empty_items_for_no_matches(self, client: AsyncClient) ->
...
+        limit: int,
+    ) -> None:
+        workspace_id = await _create_workspace(client, "limit validation")
+
+        response = await client.get(
+            f"/v1/workspaces/{workspace_id}/events",
+            params={"limit": limit},
+        )
+
+        assert response.status_code == 422
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/api/test_events.py:174]: fixed in commit 42293d47affd1066d2f46226efe53eb88f179d5e

File: src/awf/api/routes/workspaces.py [hunk 4]
```diff
@@ -211,6 +212,31 @@ async def list_workspace_overview(
...
+        )
+
+    rows = await WorkspaceEventRepository(session).list(
+        workspace_id=workspace_id,
+        event_type=event_type,
+        limit=limit,
+    )
+    return WorkspaceEventListResponse(
+        items=[WorkspaceEventResponse.model_validate(row) for row in rows]
+    )
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/api/routes/workspaces.py:236]: fixed in commit 08438386aac573c006301c9b3c75fe1d9357bd2b
